src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1004,52 +906,33 @@ private static Node RotateRightLeft(Node node)
         }
 
         /// <summary>
-        /// Testing counter that can track rotations
+        /// Testing counter that can track rotations.
         /// </summary>
         private static TreeRotation RotationNeeded(Node parent, Node current, Node sibling)
         {
-            Debug.Assert(IsRed(sibling.Left) || IsRed(sibling.Right), "sibling must have at least one red child");
-            if (IsRed(sibling.Left))
-            {
-                if (parent.Left == current)
-                {
-                    return TreeRotation.RightLeftRotation;
-                }
-                return TreeRotation.RightRotation;
-            }
-            else
-            {
-                if (parent.Left == current)
-                {
-                    return TreeRotation.LeftRotation;
-                }
-                return TreeRotation.LeftRightRotation;
-            }
+            Debug.Assert(IsRed(sibling.Left) || IsRed(sibling.Right), "The sibling must have at least one red child.");
+            bool currentIsLeft = parent.Left == current;
+            return IsRed(sibling.Left) ?
+                (currentIsLeft ? TreeRotation.RightLeft : TreeRotation.Right) :
+                (currentIsLeft ? TreeRotation.Left : TreeRotation.LeftRight);
         }
 
         /// <summary>
-        /// Used for deep equality of SortedSet testing
+        /// Used for deep equality of SortedSet testing.
         /// </summary>
-        /// <returns></returns>
-        public static IEqualityComparer<SortedSet<T>> CreateSetComparer()
-        {
-            return new SortedSetEqualityComparer<T>();
-        }
+        public static IEqualityComparer<SortedSet<T>> CreateSetComparer() => CreateSetComparer(memberEqualityComparer: null);
```

Comment:
@ianhays It allows us to delete the default constructor. See https://github.com/dotnet/corefx/pull/15376/files/679f0ecf3ffd72e1ce8c59dfd193efc8ae2a65d0#diff-69184148cd09d8a2165cb984cc974910L2636